package.json

```diff
@@ -2,14 +2,18 @@
   "name": "eslint-plugin-bpmn-io",
   "version": "1.0.1",
   "description": "Shared lint rules for bpmn.io projects",
-  "main": "index.js",
+  "type": "module",
```

Comment:
Let's add a breaking change to commit.

Reply:
👍🏻

Reply:
Updated the commit message to:

### feat: migrate to ESLint 9 and flat configs

BREAKING CHANGE: This plugin is now an ES module compatible with ESLint 9.

- Node.js version that supports ES modules is now required
- ESLint 9.0.0 is now required
- Import statements must be used instead of require() in ES module projects
- dynamic import() or the @eslint/compat package must be used in CommonJS projects

Closes https://github.com/bpmn-io/eslint-plugin-bpmn-io/issues/19
Related to https://github.com/bpmn-io/internal-docs/issues/1042

Reply:
@barmac @nikku Can you have a look?

Reply:
Add: In a commonjs project, the eslint configuration needs to use `.mjs` extension to be recognized by Node.

Reply:
But looks good overall

Reply:
I see Nico is making some changes too

Reply:
Cf. https://github.com/bpmn-io/eslint-plugin-bpmn-io/pull/20/commits/b1362f3643d679043dfa53f1d3fdd41122c2edc5, https://github.com/bpmn-io/eslint-plugin-bpmn-io/pull/20/commits/dd859631f875fd1e5b089a67cf96298deb408ca6, https://github.com/bpmn-io/eslint-plugin-bpmn-io/pull/20/commits/e14ca88628df70613aed03b9227e55f38ed92c6c.

I think we're good to merge this.